### ISSUE TITLE
Incorrect results when select uses measurement regex without field regex

### ISSUE THREAD

**Issue by dmitrios** [influxdata/influxdb#7877]
### Bug report

__System info:__ [1.1.2, MacOs, 10.12.3] 

__Steps to reproduce:__

Two measurements, M1 and M2 with same fields: A and B. These are fields, not tags. 

1. select "A","B" from /M1|M2/ =>empty
2. select A,/B/ from /M1|M2/ => correct result for B
3. select /A/,/B/ from /M1|M2/ =>correct result A,B

__Expected behavior:__ [the value of the fields "A" and "B" are correctly selected in all queries]

__Actual behavior:__ [it appears that regex in field expressions TRIGGERS parsing regex in measurement expression]

__Additional info:__ [the query 1 works fine with influx 1.1.0-1 on Windows and MacOs] This new behavior in influxdb 1.2.0-1 will break Grafana.

**Comment by e-dard** [comment]
Version `1.1.0`:

```
>
> create database "db"
> insert m1 a=1,b=2
> insert m2 a=1,b=2
>
>
> select "a","b" from /m1|m2/
name: m1
time                a b
----                - -
1485354751060183470 1 2

name: m2
time                a b
----                - -
1485354752738582585 1 2

>
>
>
> select a,/b/ from /m1|m2/
name: m1
time                a b
----                - -
1485354751060183470 1 2

name: m2
time                a b
----                - -
1485354752738582585 1 2

>
>
> select /a/,/b/ from /m1|m2/
name: m1
time                a b
----                - -
1485354751060183470 1 2

name: m2
time                a b
----                - -
1485354752738582585 1 2

>
```

Version `1.2.0`:

```
> create database "db"
> insert m1 a=1,b=2
> insert m2 a=1,b=2
>
> select "a","b" from /m1|m2/
>
>
>
> select a,/b/ from /m1|m2/
name: m1
time                a b
----                - -
1485354644496814831   2

name: m2
time                a b
----                - -
1485354651667789584   2

>
>
> select /a/,/b/ from /m1|m2/
name: m1
time                a b
----                - -
1485354644496814831 1 2

name: m2
time                a b
----                - -
1485354651667789584 1 2

>
```

In the case of query number (2), notice that the field `a` shows up for `1.2.0` but it's not populated.

**Comment by e-dard** [comment]
A cursory glance of `statement_executor.go` changes indicates that it probably has something to do with this change: https://github.com/influxdata/influxdb/commit/d7c8c7ca4fbb86ac95d1828aaed82545a3d70883#diff-084cb29c383c1c0fffae2fac08113ad5L547 introduced in #7646.

**Comment by jsternberg** [comment]
I think I know what this issue might be. I think what's happening is the part of the code that maps types isn't working with regexes. But when you do `/a/` for a field, it retrieves the type as part of expanding the field regex so it doesn't happen when you use that.

I'll confirm and fix it.

**Comment by jsternberg** [comment]
Fixed via #7881.
